pkg/opensearch/response_parser.go

```diff
@@ -588,6 +608,25 @@ func (rp *responseParser) getMetricName(metric string) string {
 	return metric
 }
 
+func castToFloat(j *simplejson.Json) *float64 {
```

Comment:
The originally-used castToNullFloat function below uses an underlying standard library type sql.NullFloat64 which is also a way to distinguish between something "null/nil" or with a float64 value. In this case, we can also just use a *float64 type to distinguish between "nil" and float64 and there is no strong reason to use an imported type for that. 

Here's a similar discussion: https://stackoverflow.com/questions/33458439/using-sql-nullfloat64-vs-float64-in-golang

In Elasticsearch I noticed they moved towards *float64 so I copied that; I think that in further refactors the castToNullFloat will no longer be used.